tm2/pkg/bft/node/node.go

```diff
@@ -193,35 +194,35 @@ func createAndStartProxyAppConns(clientCreator proxy.ClientCreator, logger log.L
 	return proxyApp, nil
 }
 
-func createAndStartIndexerService(config *cfg.Config, dbProvider DBProvider,
-	evsw events.EventSwitch, logger log.Logger,
+func createAndStartIndexerService(
+	cfg *cfg.Config,
+	evSwitch events.EventSwitch,
+	logger log.Logger,
 ) (*txindex.IndexerService, txindex.TxIndexer, error) {
-	var txIndexer txindex.TxIndexer = &null.TxIndex{}
-	/*
-		switch config.TxIndex.Indexer {
-		case "kv":
-			store, err := dbProvider(&DBContext{"tx_index", config})
-			if err != nil {
-				return nil, nil, err
-			}
-			switch {
-			case config.TxIndex.IndexTags != "":
-				txIndexer = kv.NewTxIndex(store, kv.IndexTags(splitAndTrimEmpty(config.TxIndex.IndexTags, ",", " ")))
-			case config.TxIndex.IndexAllTags:
-				txIndexer = kv.NewTxIndex(store, kv.IndexAllTags())
-			default:
-				txIndexer = kv.NewTxIndex(store)
-			}
-		default:
-			txIndexer = &null.TxIndex{}
+	var (
+		err       error
+		txIndexer txindex.TxIndexer
+	)
+
+	// Instantiate the indexer based on the configuration
+	switch cfg.Indexer.IndexerType {
+	case file.IndexerType:
+		// Transaction indexes should be logged to files
+		txIndexer, err = file.NewTxIndexer(cfg.Indexer)
+		if err != nil {
+			return nil, nil, fmt.Errorf("unable to create file tx indexer, %w", err)
 		}
-	*/
+	default:
+		// Transaction indexing should be omitted
+		txIndexer = null.NewNullIndexer()
+	}
 
-	indexerService := txindex.NewIndexerService(txIndexer, evsw)
+	indexerService := txindex.NewIndexerService(txIndexer, evSwitch)
```

Comment:
it's evsw everywhere.

Reply:
Reverted:

[16f07c8](https://github.com/gnolang/gno/pull/546/commits/16f07c81b15480b090f16a90e5e38b718372b63a)